Octokit/Models/Request/Enterprise/UpdateMaintenanceRequest.cs

```diff
@@ -0,0 +1,107 @@
+        /// </summary>
+        public UpdateMaintenanceRequest(UpdateMaintenanceRequestDetails maintenance)
+        {
+            Ensure.ArgumentNotNull(maintenance, "maintenance");
+
+            Maintenance = maintenance;
+        }
+
+        /// <summary>
+        /// Details for the maintenance request
+        /// </summary>
+        public UpdateMaintenanceRequestDetails Maintenance { get; protected set; }
+
+        internal string DebuggerDisplay
+        {
+            get
+            {
+                return string.Format(CultureInfo.InvariantCulture, "Maintenance: {0}", this.Maintenance.DebuggerDisplay);
+            }
+        }
+    }
+
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class UpdateMaintenanceRequestDetails
+    {
+        /// <summary>
+        /// Maintenance request details with default values (results in Maintenance mode being turned off immediately)
+        /// </summary>
+        public UpdateMaintenanceRequestDetails()
+        { }
+
+        /// <summary>
+        /// Maintenance request details to enable/disable maintenance mode immediately
+        /// </summary>
+        /// <param name="enabled">true to enable, false to disable</param>
+        public UpdateMaintenanceRequestDetails(bool enabled)
+        {
+            Enabled = enabled;
+            When = "now";
+        }
+
+        /// <summary>
+        /// Maintenance request details to enable/disable maintenance at a specified time (only applicable when enabling maintenance)
+        /// </summary>
+        /// <param name="enabled">true to enable, false to disable</param>
+        /// <param name="when">A phrase specifying when maintenance mode will be enabled.  Phrase uses humanised forms supported by the <a href="https://github.com/mojombo/chronic">mojombo/chronic library</a> used by the GitHub API
+        /// such as "this friday at 5pm" or "5 minutes before midday tomorrow"
+        /// <remarks>If enabled is false, the when parameter is ignored and maintenance is turned off immediately</remarks>
+        public UpdateMaintenanceRequestDetails(bool enabled, string when)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(when, "when");
+
+            Enabled = enabled;
+            When = when;
+        }
+
+        /// <summary>
+        /// Maintenance request details to enable/disable maintenance at a specified time (only applicable when enabling maintenance)
+        /// </summary>
+        /// <param name="enabled">true to enable, false to disable</param>
+        /// <param name="when">A <see cref="DateTimeOffset"/> specifying when maintenance mode will be enabled.</param>
+        /// <remarks>If enabled is false, the when parameter is ignored and maintenance is turned off immediately</remarks>
+        public UpdateMaintenanceRequestDetails(bool enabled, DateTimeOffset when)
+        {
+            Enabled = enabled;
+            When = when.ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ssZ",
```

Comment:
> I'll see if the GHE API is happy to just receive the out of the box .ToString("o") format

@khellang : No dice! :stuck_out_tongue: 

An internal server error 500 is thrown when using this format.

All of these values cause an error
```
2016-08-16T21:15:06.9332018+00:00 // ToUniversalTime().ToString("o")
2016-08-17T07:05:06.9332018+10:00 // .ToString("o")
2016-08-16T21:15:06+00:00 // UTC with custom format removing the fffffff component
2016-08-17T07:05:06+10:00 // Local time with custom format removing the fffffff component
2016-08-16T21:15:06+0000 // UTC with custom format removing the fffffff component and removing colon in time zone
2016-08-17T07:05:06+1000 // Local time with custom format removing the fffffff component and removing colon in time zone
2016-08-16T21:15:06+00 // UTC with custom format removing the fffffff component and only 2 digits for time zone
2016-08-17T07:05:06+10 // Local time with custom format removing the fffffff component and only 2 digits for time zone
```
These formats work:
```
`2016-08-16T21:15:06Z` // .ToUniversalTime() with custom format using a static "Z" for zulu timezone
`2016-08-16T21:15:06.9332018Z` // .ToUniversalTime() with custom format including fffffff and using a static "Z" for zulu timezone
```

So yeah basically it's the timezone designator that the GHE API seems to only accept when it is "Z" and not any of the other (valid ISO8601) values.